legend-engine-ide-lsp-default-extensions/src/main/java/org/finos/legend/engine/ide/lsp/extension/core/FunctionExecutionSupport.java

```diff
@@ -192,6 +309,181 @@ static void executePlan(GlobalState globalState, FunctionExecutionSupport execut
+
+            boolean debug = Boolean.parseBoolean(executableArgs.get("debug"));
+
+            String result = objectMapper.writeValueAsString(debug ?
+                    FunctionExecutionSupport.debugExecutionPlan(lambda, mappingPath, runtime, context, pureModel, clientVersion) :
+                    FunctionExecutionSupport.getExecutionPlan(lambda, mappingPath, runtime, context, pureModel, clientVersion));
+            results.add(
+                    FunctionLegendExecutionResult.newResult(
+                            entityPath,
+                            LegendExecutionResult.Type.SUCCESS,
+                            result,
+                            null,
+                            section.getDocumentState().getDocumentId(),
+                            section.getSectionNumber(),
+                            inputParameters
+                    )
+            );
+        }
+        catch (Exception e)
+        {
+            results.add(extension.errorResult(e, entityPath));
+        }
+        return results;
+    }
+
+    static Iterable<? extends LegendExecutionResult> convertGrammarToJSON_lambda(FunctionExecutionSupport executionSupport, SectionState section, String entityPath, Map<String, String> executableArgs, Map<String, Object> inputParameters)
+    {
+        AbstractLSPGrammarExtension extension = executionSupport.getExtension();
+
+        MutableList<LegendExecutionResult> results = Lists.mutable.empty();
+        try
+        {
+            String sourceId = executableArgs.getOrDefault("sourceId", "");
+            int lineOffset = Integer.parseInt(executableArgs.getOrDefault("lineOffset", "0"));
+            int columnOffset = Integer.parseInt(executableArgs.getOrDefault("columnOffset", "0"));
+            boolean returnSourceInformation = Boolean.parseBoolean(executableArgs.getOrDefault("returnSourceInformation", "true"));
+            Lambda lambda = PureGrammarParser.newInstance().parseLambda(executableArgs.get("code"), sourceId, lineOffset, columnOffset, returnSourceInformation);
+            results.add(
+                    FunctionLegendExecutionResult.newResult(
+                            entityPath,
+                            LegendExecutionResult.Type.SUCCESS,
+                            objectMapper.writeValueAsString(lambda),
+                            null,
+                            section.getDocumentState().getDocumentId(),
+                            section.getSectionNumber(),
+                            inputParameters
+                    )
+            );
+        }
+        catch (Exception e)
+        {
+            results.add(extension.errorResult(e, entityPath));
+        }
+        return results;
+    }
+
+    static Iterable<? extends LegendExecutionResult> convertJSONToGrammar_lambda(FunctionExecutionSupport executionSupport, SectionState section, String entityPath, Map<String, String> executableArgs, Map<String, Object> inputParameters)
```

Comment:
Do we need this method and the one below?  Can we just use the bulk even for single calls?

If we end needing this method, can we delegate to the bulk one (or vice-versa) and avoid some duplication?  Also, lets name it following Java convention: `convertLambdaJsonToGrammar`

Reply:
Good point. I removed the single version and kept only the batch version. The client will just call the batch version with a map with one element if it only needs to convert 1 JSON.